lib/kubeclient/application_default_credentials.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require 'googleauth'
```

Comment:
With this file required in `kubeclient.rb`, I think this means you can't require kubeclient without having googleauth gem.  Can you move this into `token` method so it's only done if you use it?

Reply:
Good catch. Fixed in 774177b.